datalad/support/sshconnector.py

```diff
@@ -521,6 +521,12 @@ def open(self):
         ConnectionOpenFailedError
           When starting the SSH ControlMaster process failed.
         """
+        with fasteners.process_lock.InterProcessLock(self.ctrl_path.with_suffix('.lck')):
```

Comment:
Could `datalad.support.locking.lock_if_check_fails` be used here to avoid creating a lock in many cases?

Reply:
oh right 
1. we should just have a lock instance bound to the instance and reuse it (should shave off some cycles).  
2. Indeed we should be able to use `lock_if_check_fails` but it would loose that (reuse instance) so I might need to RF it to allow to reuse a lock provided from outside to take advantage of `1.`

Reply:
Decided against using `lock_if_check_fails`: it was coded up with use cases in mind where check would check for something you expect to not disappear (e.g. downloaded annex key for an archive) between you checking and then doing some next action... so for faster paced use cases I am afraid there might be a race condition that we would happily not bother establishing the socket while it disappears from us.  So, to be safer than sorry,

I think there might still be benefit from reusing a lock instance so I have pushed a RF which does that while relying on a feature of fasteners.locked to have a list of locks instead of a single one.  So the diff becomes shorter ;)